cmd/kubelet/app/options/options.go

```diff
@@ -344,6 +344,7 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
 			"v":                   true,
 			"vmodule":             true,
 			"log-flush-frequency": true,
+			"provider-id":         true,
```

Comment:
I would also consider undeprecating --node-ip

Reply:
Agree, the EKS AMI (amazon machine image) bootstrap script currently sets `--node-ip` ([ref](https://github.com/awslabs/amazon-eks-ami/blob/master/files/bootstrap.sh#L458)) as well as a number of other kubelet flags and/or config options, which are clunky to set in a script.  [See here](https://github.com/awslabs/amazon-eks-ami/blob/master/files/bootstrap.sh#L465) for how we currently set provider ID for reference.

Reply:
>      --node-ip string                                           IP address (or comma-separated dual-stack IP addresses) of the node. If unset, kubelet will use the node's default IPv4 address, if any, or its default IPv6 address if it has no IPv4 addresses. You can pass '::' to make it prefer the default IPv6 address rather than the default IPv4 address.

`node-ip` is not deprecated. `nodeIP` is not supported kubelet configuration. `providerID`is in kubelet configuration and it can be used instead of `--provider-id`.

https://github.com/kubernetes/kubelet/blob/7590f9e6307fa2a6d3e377b7bf54e06cdc29f5fe/config/v1beta1/types.go#L677